models/user.go

```diff
@@ -520,5 +520,46 @@ func (u *User) IsBanned() bool {
 
 func (u *User) UpdateBannedUntil(tx *storage.Connection) error {
 	return tx.UpdateOnly(u, "banned_until")
+}
+
+// RemovePassword removes the user's password and email provider information
+func (u *User) RemovePassword(tx *storage.Connection) error {
+	u.EncryptedPassword = ""
+
+	metadata := u.AppMetaData
+
+	if metadata != nil {
+		provider := metadata["provider"]
+		providers := metadata["providers"]
+
+		_, ok := provider.(string)
+
+		if provider != nil && ok && provider.(string) == "email" {
+			delete(metadata, "provider")
```

Comment:
Not sure about this -- would this delete the OAuth provider now? Need to confirm.

Reply:
Actually -- clearly no, but what will the provider be?

Reply:
Actually the `provider` field is going to be deprecated - it represents the first provider the user used to sign up with. We created the `providers` field (note the "s") to keep track of all the providers associated with a user's account subsequently.